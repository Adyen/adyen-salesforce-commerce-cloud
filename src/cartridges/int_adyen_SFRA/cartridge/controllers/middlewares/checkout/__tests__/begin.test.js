/* eslint-disable global-require */
const {
  checkout: { begin },
} = require('../../index');
const OrderMgr = require('dw/order/OrderMgr');
const BasketMgr = require('dw/order/BasketMgr');
const ShippingMgr = require('dw/order/ShippingMgr');
const Logger = require('dw/system/Logger');
const Transaction = require('dw/system/Transaction');

let res;
let req;
beforeEach(() => {
  jest.clearAllMocks();
  req = {
    currentCustomer: { raw: { isAuthenticated: jest.fn(() => false) } },
    session: { privacyCache: { get: jest.fn(), set: jest.fn() } },
  };
  res = { getViewData: jest.fn(() => ({})), setViewData: jest.fn() };
});

describe('Begin', () => {
  it('should update saved cards', () => {
    const {
      updateSavedCards,
    } = require('*/cartridge/scripts/updateSavedCards');
    req.currentCustomer.raw.isAuthenticated.mockImplementation(() => true);
    begin(req, res, jest.fn());
    expect(updateSavedCards).toBeCalledTimes(1);
  });
  it('should set view data', () => {
    begin(req, res, jest.fn());
    expect(res.setViewData.mock.calls).toMatchSnapshot();
  });

  it('should not attempt to restore cart when no order number is cached', () =>{
    begin(req, res, jest.fn());
    expect(res.setViewData.mock.calls).toMatchSnapshot();
    expect(OrderMgr.failOrder).not.toHaveBeenCalled();
  })

  it('should not attempt to restore cart when cart is not empty', () => {
    req.session.privacyCache.get.mockImplementationOnce(() =>{ return '12312' });
    OrderMgr.status = { value: "0"};
    begin(req, res, jest.fn());
    expect(Transaction.wrap).not.toHaveBeenCalled();
  })

  it('should successfully restore cart when current cart is empty and order number is in cache', () => {
    req.session.privacyCache.get.mockImplementationOnce(() =>{ return "0" });
    BasketMgr.getAllProductLineItems.mockImplementationOnce(() =>{ return []; });
    begin(req, res, jest.fn());
    expect(Transaction.wrap).toHaveBeenCalled();
  })
});
