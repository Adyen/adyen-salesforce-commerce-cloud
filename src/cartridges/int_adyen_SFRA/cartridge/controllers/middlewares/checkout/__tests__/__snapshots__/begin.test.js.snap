// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Begin should not attempt to restore cart when no order number is cached 1`] = `
Array [
  Array [
    Object {
      "adyen": Object {
        "SFRA6Enabled": false,
        "adyenClientKey": "mocked_client_key",
        "cardholderNameBool": true,
        "clientKey": "mocked_client_key",
        "environment": "test",
        "googleMerchantID": "mocked_google_merchant_id",
        "installments": true,
        "merchantAccount": "mocked_merchant_account",
      },
    },
  ],
]
`;

exports[`Begin should set view data 1`] = `
Array [
  Array [
    Object {
      "adyen": Object {
        "SFRA6Enabled": false,
        "adyenClientKey": "mocked_client_key",
        "cardholderNameBool": true,
        "clientKey": "mocked_client_key",
        "environment": "test",
        "googleMerchantID": "mocked_google_merchant_id",
        "installments": true,
        "merchantAccount": "mocked_merchant_account",
      },
    },
  ],
]
`;
